include/clink/linalg/vector.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_LINALG_VECTOR_H_
+#define CLINK_LINALG_VECTOR_H_
+
+#include "tfrt/host_context/chain.h"
+
+namespace clink {
+
+// A vector of double values.
+class Vector {
+public:
+  virtual ~Vector() {}
```

Comment:
Should we follow the `Tensor` in TFRT as example and make the following changes:
- Remove the copy constructor.
- Add a default constructor as `protected`.
- Make the default destructor `protected`.